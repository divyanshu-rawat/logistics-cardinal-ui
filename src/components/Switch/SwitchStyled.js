import styled from 'styled-components';

const SwitchStyled = styled('div')`
  display: flex;
  align-items: center;
  label {
    margin-bottom: 0;
    margin-right: ${({ theme }) => theme.spacings.kilo};
  }
  .react-switch-bg {
    ${({ theme, active }) => {
      if (!active) {
        return `border: 1px solid ${theme.colors.grey200}`;
      }
      return '';
    }};
  }
`;

export default SwitchStyled;
