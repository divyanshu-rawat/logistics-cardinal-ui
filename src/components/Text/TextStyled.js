import styled from 'styled-components';

const textSizeVariant = (size, theme) => {
  switch (size) {
    case 'xsmall':
      return theme.textFontSizeXSmall;
    case 'small':
      return theme.textFontSizeSmall;
    case 'large':
      return theme.textFontSizeLarge;
    case 'xlarge':
      return theme.textFontSizeXLarge;
    default:
      return theme.textFontSize;
  }
};

const TextStyled = styled('span')`
  color: ${({ theme, color }) => color || theme.textColor};
  font-family: ${({ theme, fontFamily }) => fontFamily || theme.textFontFamily};
  font-weight: ${({ theme, weight }) => weight || theme.textFontWeight};
  text-align: ${({ align }) => align || 'left'};
  text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'auto')};
  margin: ${({ margin, theme }) => margin || theme.textSpacing};
  font-size: ${({ fontSize, theme }) => textSizeVariant(fontSize, theme)};
  display: ${({ block }) => (block ? 'block' : 'inline-block')};

  ${({ truncate }) => {
    if (truncate) {
      return `
        display: block;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      `;
    }

    return '';
  }};
`;

export default TextStyled;
