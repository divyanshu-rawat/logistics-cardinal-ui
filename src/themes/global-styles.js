import { createGlobalStyle } from 'styled-components';
import 'normalize.css';

export const GlobalStyles = createGlobalStyle`
  ${({ theme, injectGlobal }) => `
    *,
    *::before,
    *::after {
      box-sizing: border-box;
    }

    html {
      box-sizing: border-box;
      line-height: 1.15;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
      -ms-overflow-style: scrollbar;
      -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
      [type='button'] {
        appearance: none;
      }
    }

    body {
      margin: 0;
      background-color: ${theme.bodyBackground};
      color: ${theme.textColor};
      font-family: ${theme.textFontFamily};
      font-size: ${theme.textFontSize};
      font-weight: ${theme.textFontWeight};
      line-height: ${theme.lineHeightBase};
    }

    fieldset {
      min-width: 0;
      padding: 0;
      margin: 0;
      border: 0;
    }

    .Drawer {
      &--is-open {
        overflow: hidden;
      }
    }

    .ReactModal {
      &__Body--open {
        overflow: hidden;
      }
      &__Overlay {
        position: fixed;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        transition: opacity 200ms ease-in-out;
        height: auto;
        overflow: auto;
        z-index: ${theme.zIndex.modalOverlay};
        background-color: ${theme.overlayColor};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
      &__Content {
        margin: 10em auto;
        position: relative;
        left: auto;
        right: auto;
        top: auto;
        bottom: auto;
        transform: none;
        opacity: 0;
        transition: opacity 200ms ease-in-out;
        min-height: ${theme.modalMinHeight};
        min-width: ${theme.modalMinWidth};
        width: ${theme.modalWidth};
        max-width: ${theme.modalMaxWidth};
        background-color: ${theme.modalBackgroundColor};
        border: 1px ${theme.modalBorderStyle} ${theme.modalBorderColor};
        box-shadow: ${theme.modalBoxShadow};
        border-radius: ${theme.modalBorderRadius};
        outline: ${theme.modalOutline};
        z-index: ${theme.zIndex.modalContent};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
    }


    ${injectGlobal && injectGlobal(theme)};
  `}
`;
