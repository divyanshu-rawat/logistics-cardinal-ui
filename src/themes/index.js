export rooster from './rooster';
export standard from './standard';
export hurrier from './hurrier';
