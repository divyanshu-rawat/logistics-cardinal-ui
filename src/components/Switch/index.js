import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RCSwitch from 'react-switch';
import { withTheme } from 'styled-components';

import SwitchStyled from './SwitchStyled';

import { SIZES } from '../utils/constants';

const DIMENSIONS = {
  sm: {
    width: 40,
    height: 22,
  },
  md: {
    width: 80,
    height: 42,
  },
  lg: {
    width: 110,
    height: 62,
  },
};

class Switch extends Component {
  static propTypes = {
    /** current theme -- by context */
    theme: PropTypes.object.isRequired,
    /** initial value */
    initialValue: PropTypes.bool.isRequired,
    /** onChange callback */
    onChangeStatus: PropTypes.func.isRequired,
    /** Dimensions */
    size: PropTypes.oneOf(Object.keys(SIZES)),
    /** HTML id */
    id: PropTypes.string,
    /** Receives (active) param (active) => active ? 'on' : 'off' */
    labelText: PropTypes.func,
    /** Switcher ON and OFF colors */
    colors: PropTypes.shape({
      on: PropTypes.string,
      off: PropTypes.string,
    }),
  };

  static defaultProps = {
    size: 'sm',
    initialValue: false,
    colors: {},
  };

  state = {
    active: this.props.initialValue,
  };

  onChangeSwitch = (value) => {
    const { onChangeStatus } = this.props;
    this.setState({ active: !this.state.active }, () => onChangeStatus(value));
  };

  render() {
    const { theme, size, colors, labelText, id } = this.props;
    const { active } = this.state;
    const onColor = colors.on || theme.colors.primary100;
    const offColor = colors.off || theme.colors.grey400;
    const { width, height } = DIMENSIONS[size];

    return (
      <SwitchStyled active={active}>
        {labelText && <label htmlFor={id}>{labelText(active)}</label>}
        <RCSwitch
          id={id}
          height={height}
          width={width}
          onColor={onColor}
          offColor={offColor}
          onChange={this.onChangeSwitch}
          checked={active}
          activeBoxShadow={theme.boxShadow.switchHandler}
          className="react-switch"
          uncheckedIcon={false}
          checkedIcon={false}
        />
      </SwitchStyled>
    );
  }
}

export default withTheme(Switch);
