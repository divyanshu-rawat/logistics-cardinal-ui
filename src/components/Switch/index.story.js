import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { boolean } from '@storybook/addon-knobs';

import withTests from '../../utils/withTests';

import { FORM } from '../../../.storybook/sections';

import Switch from '.';
import Text from '../Text';
import Themes from '../../themes';

storiesOf(`${FORM}Switch`, module)
  .addDecorator(withTests('Switch/tests/index', 'Switch/tests/SwitchStyled'))
  .add(
    'Simple',
    withInfo()(() => (
      <Switch
        theme={Themes.rooster}
        initialValue={false}
        onChangeStatus={action((item) => console.log(item))}
      />
    )),
  )
  .add(
    'With Label',
    withInfo()(() => (
      <Switch
        initialValue={false}
        theme={Themes.rooster}
        id="myAwesomeSwitch"
        labelText={(active) =>
          active ? (
            <Text margin="0">The switch is on.</Text>
          ) : (
            <Text margin="0">The switch is off.</Text>
          )
        }
        onChangeStatus={action((item) => console.log(item))}
      />
    )),
  )
  .add(
    'With initial value',
    withInfo()(() => (
      <Switch
        theme={Themes.rooster}
        initialValue
        onChangeStatus={action((item) => console.log(item))}
      />
    )),
  )
  .add(
    'Disabled',
    withInfo()(() => (
      <Switch
        theme={Themes.rooster}
        initialValue
        disabled={boolean('disabled', true)}
        onChangeStatus={action((item) => console.log(item))}
      />
    )),
  )
  .add(
    'Size small',
    withInfo()(() => (
      <Switch
        theme={Themes.rooster}
        initialValue
        size="sm"
        onChangeStatus={action((item) => console.log(item))}
      />
    )),
  )
  .add(
    'Size medium',
    withInfo()(() => (
      <Switch
        theme={Themes.rooster}
        initialValue
        size="md"
        onChangeStatus={action((item) => console.log(item))}
      />
    )),
  )
  .add(
    'Size large',
    withInfo()(() => (
      <Switch
        initialValue
        theme={Themes.rooster}
        size="lg"
        onChangeStatus={action((item) => console.log(item))}
      />
    )),
  );
