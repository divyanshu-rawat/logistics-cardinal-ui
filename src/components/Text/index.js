import React from 'react';
import PropTypes from 'prop-types';
import TextStyled from './TextStyled';

const Text = ({ children, ...props }) => {
  const RenderAsProp = TextStyled.withComponent(props.as);
  return <RenderAsProp {...props}>{children}</RenderAsProp>;
};

Text.propTypes = {
  children: PropTypes.any.isRequired,
  /** Final Element HTML to be rendered */
  as: PropTypes.string.isRequired,
  /** Text Color */
  color: PropTypes.string,
  /** Font Family, default is the theme one */
  fontFamily: PropTypes.string,
  /** Font Weight, default is the theme one */
  weight: PropTypes.number,
  /** Text Align, default is left */
  align: PropTypes.string,
  /** Truncate... */
  truncate: PropTypes.bool,
  /** Text Transform, UPPERCASE */
  uppercase: PropTypes.bool,
  margin: PropTypes.string,
};

Text.defaultProps = {
  children: '',
  as: 'span',
};

export default Text;
