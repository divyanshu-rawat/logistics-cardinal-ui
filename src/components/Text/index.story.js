import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import centered from '@storybook/addon-centered';
import {
  withKnobs,
  color,
  boolean,
  select,
  number,
} from '@storybook/addon-knobs';

import withTests from '../../utils/withTests';
import Theme from '../../themes/rooster';

import Text from '.';

storiesOf('Text', module)
  .addDecorator(withTests('Text/tests/index', 'TextStyled'))
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .add(
    'Simple',
    withInfo()(() => (
      <div style={{ width: '500px' }}>
        <Text
          color={color('Color', Theme.fontFamily.color)}
          fontFamily={select('Font Family', [
            Theme.fontFamily.default,
            'Times New Roman',
          ])}
          weight={number('FontWeight', Theme.fontWeight.regular)}
          align={select('Text Align', ['left', 'right', 'center'], 'left')}
          truncate={boolean('Truncate', false)}
          uppercase={boolean('Uppercase', false)}
        >
          Mussum Ipsum, cacilds vidis litro abertis. Em pé sem cair, deitado sem
          dormir, sentado sem cochilar e fazendo pose. Praesent malesuada urna
          nisi, quis volutpat erat hendrerit non. Nam vulputate dapibus.
          Detraxit consequat et quo num tendi nada. Interagi no mé, cursus quis,
          vehicula ac nisi.
        </Text>
      </div>
    )),
  );
