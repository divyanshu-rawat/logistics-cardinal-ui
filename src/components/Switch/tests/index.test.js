import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Switch from '..';
import Text from '../../Text';
import { rooster } from '../../../themes';

const onChangeFn = jest.fn();

const renderedComponent = ({
  id,
  initialValue = false,
  size = 'sm',
  labelText,
  colors,
  disabled = false,
}) =>
  shallow(
    <Switch
      id={id}
      theme={rooster}
      disabled={disabled}
      initialValue={initialValue}
      size={size}
      labelText={labelText}
      onChangeStatus={onChangeFn}
      colors={colors}
    />,
  );

describe('<Switch />', () => {
  it('should render the component', () => {
    const tree = toJson(renderedComponent({}));
    expect(tree).toMatchSnapshot();
  });

  it('Switch: should render component with given props', () => {
    const props = {
      id: 'someId',
      initialValue: true,
      size: 'md',
      labelText: () => <Text>Some Text</Text>,
      disabled: true,
      colors: {
        on: '#f00',
        off: '#d5d5d5',
      },
    };
    const tree = toJson(renderedComponent(props).dive());
    expect(tree).toMatchSnapshot();
  });

  it('Switch: should call `onChangeFn` when value is changed', () => {
    const component = renderedComponent({}).dive();

    component
      .find('ReactSwitch')
      .simulate('change', { target: { value: true } });

    expect(onChangeFn.mock.calls.length).toBe(1);
  });
});
