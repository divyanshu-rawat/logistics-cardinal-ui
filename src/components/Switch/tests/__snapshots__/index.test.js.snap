// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> Switch: should render component with given props 1`] = `
<styled.div
  active={true}
>
  <label
    htmlFor="someId"
  >
    <Text
      as="span"
    >
      Some Text
    </Text>
  </label>
  <ReactSwitch
    activeBoxShadow="0px 0px 1px 5px rgba(255, 255, 255, 0.2)"
    boxShadow={null}
    checked={true}
    checkedIcon={false}
    className="react-switch"
    disabled={true}
    height={42}
    id="someId"
    offColor="#d5d5d5"
    offHandleColor="#fff"
    onChange={[Function]}
    onColor="#f00"
    onHandleColor="#fff"
    uncheckedIcon={false}
    width={80}
  />
</styled.div>
`;

exports[`<Switch /> should render the component 1`] = `
<Switch
  colors={Object {}}
  disabled={false}
  initialValue={false}
  onChangeStatus={[Function]}
  size="sm"
  theme={
    Object {
      "borderRadius": Object {
        "giga": "5px",
        "kilo": "1px",
        "mega": "4px",
      },
      "borderWidth": Object {
        "bit": "1px",
        "byte": "2px",
      },
      "boxShadow": Object {
        "switchHandler": "0px 0px 1px 5px rgba(255, 255, 255, 0.2)",
      },
      "breakpoints": Object {
        "default": Object {
          "minWidth": 768,
        },
        "desktop": Object {
          "maxWidth": 1200,
          "minWidth": 992,
        },
        "largeDesktop": Object {
          "minWidth": 1200,
        },
        "mobile": Object {
          "maxWidth": 768,
          "minWidth": 0,
        },
        "tablet": Object {
          "maxWidth": 992,
          "minWidth": 768,
        },
      },
      "colors": Object {
        "black": "#000",
        "bodyBackground": "#292d33",
        "danger": "#a94442",
        "green100": "#c6e5cb",
        "green300": "#97e2a3",
        "green500": "#62de76",
        "green700": "#49b85b",
        "green900": "#2f903e",
        "grey100": "#96a2b0",
        "grey200": "#3f454d",
        "grey300": "#343a42",
        "grey400": "#292d33",
        "grey500": "#40464f",
        "mainBackground": "343941",
        "primary100": "#3f9bfd",
        "primary200": "#375f8a",
        "primary300": "#324e70",
        "primary400": "#96a2b0",
        "primary500": "#484e58",
        "red100": "#ffe6e1",
        "red300": "#ffaf9f",
        "red500": "#ff7559",
        "red700": "#d55a41",
        "red900": "#a94442",
        "success": "#49b85b",
        "textColor": "#fff",
        "warning": "#d4a546",
        "white": "#fff",
        "yellow100": "#f2e5cb",
        "yellow300": "#f7da9f",
        "yellow500": "#ffc859",
        "yellow700": "#d4a546",
        "yellow900": "#ab8433",
      },
      "composedSpacings": Object {
        "byte": "10px 15px",
      },
      "fontFamily": Object {
        "color": "#fff",
        "default": "\\"Open Sans\\", sans-serif",
      },
      "fontSize": Object {
        "bit": "11px",
        "byte": "13px",
        "giga": "24px",
        "kilo": "15px",
        "mega": "18px",
      },
      "fontWeight": Object {
        "bold": 700,
        "regular": 400,
        "semiBold": 600,
      },
      "grid": Object {
        "outerGutter": "10px",
      },
      "lineHeight": Object {
        "bit": 1.153,
        "byte": 1.333,
        "kilo": 1.5,
      },
      "spacings": Object {
        "bit": "1px",
        "byte": "5px",
        "giga": "20px",
        "kilo": "10px",
        "mega": "15px",
      },
      "zIndex": Object {
        "navbar": 2030,
        "select": 1090,
        "spinner": 2010,
        "spinnerOverlay": 2000,
        "tooltip": 2020,
      },
    }
  }
/>
`;
